### PR TITLE
Add missing Rotation functions

### PR DESCRIPTION
**Pull request by chrisflesher**
Added some missing functions so the API matches scipy now:
  - `align_vectors`
  - `create_group`
  - `random`
  - `reduce`

Also fixed a bug in `from_euler` when only one axis is specified.

### CODE REVIEW

**Review by jonesk7734 — APPROVED** [review]
😀

**Comment by chrisflesher** [comment]
@jonesk7734 thanks for reviewing. Made another attempt to get the unit tests working. Could you re-run CI? Thanks!

**Review by jonesk7734 — APPROVED** [review]
😃

**Review by jakevdp — CHANGES_REQUESTED** [review]
Looks good, though it seems there are still sign degeneracy issues with the tests that will need to be addressed.

**Comment by chrisflesher** [comment]
@jonesk7734 @jakevdp Found the problem was due to small floating point errors when comparing rotations that are 360 degrees offset. The workaround I selected was to replace `jax.test_util._CheckAgainstNumpy` with very similar function called `_CheckQuaternionAgainstNumpy` that compares quaternions using `|q1⋅q2| > 1 - eps`

**Comment by chrisflesher** [comment]
I see JAX is now on version 0.4.16 but wasn't sure how to update the `CHANGELOG.md` file... Should I just let you guys do that?

**Comment by jakevdp** [comment]
The next release will be JAX v0.4.16, so any changes that have not yet been merged (like this PR) should be listed under v0.4.16.

**Comment by chrisflesher** [comment]
Hmm... I tried updating the `CHANGELOG.md` by rebasing this branch onto latest `main` but couldn't figure out how to do it properly... so am closing this PR and starting a new one.
